app/errors/services/remove_more_products_than_were_added_error.rb

```diff
@@ -0,0 +1,10 @@
+module Services
+  class RemoveMoreProductsThanWereAddedError < StandardError
+    def initialize(data)
```

Comment:
i think you could define the method like:
```suggestion
    def initialize(quantity:, product_id:)
```

Reply:
Updated.